lib/graphql/schema/object.rb

```diff
@@ -69,47 +71,70 @@ def initialize(object, context)
       end
 
       class << self
-        def implements(*new_interfaces)
+        def implements(*new_interfaces, **options)
+          new_memberships = []
           new_interfaces.each do |int|
             if int.is_a?(Module)
               unless int.include?(GraphQL::Schema::Interface)
                 raise "#{int} cannot be implemented since it's not a GraphQL Interface. Use `include` for plain Ruby modules."
               end
 
+              new_memberships << int.type_membership_class.new(int, self, options)
+
               # Include the methods here,
               # `.fields` will use the inheritance chain
               # to find inherited fields
               include(int)
+            elsif int.is_a?(String) || int.is_a?(GraphQL::Schema::LateBoundType)
+              new_memberships << int
```

Comment:
TODO options are silently ignored here.

Reply:
Maybe this is OK for now: the `type_membership_class` comes from the _interface_, and we don't have one yet here. 

This _could_ be implemented. Eg, add a `PendingTypeMembership` class which holds the pending type and the options, and then when the type is updated, add the options into the new TypeMembership class.

Reply:
Is the issue that a `type_membership_class` may not exist on the interface yet? I would assume the interface would have to be loaded and have its membership class loaded in order to use it?

Reply:
Yup